src/AutorestSwift/View Models/KeyValueViewModel.swift

```diff
@@ -32,15 +32,32 @@ import Foundation
 struct KeyValueViewModel {
     let key: String
     let value: String
+    let valueNilable: Bool
 
-    init(from param: Parameter, with model: CodeModel) {
-        self.key = param.serializedName!
-        if let constant = param.schema as? ConstantSchema {
-            self.value = "\"\(constant.value.value)\""
+    init(from param: Parameter, with model: CodeModel, and operation: Operation, using key: String? = nil) {
+        self.key = key ?? param.serializedName!
```

Comment:
What is the case where we know the key to be able to just pass it in? Also, instead of using `!` it would make more sense to have a guard statement with a `fatalError(message)`. This will crash if `serializedName` is nil, but the fatalError will yield a better error message.